### PR TITLE
Add validations to User, Reservation, Flat models, and routes

### PR DESCRIPTION
**Pull request by reba-g**
- Add validations to User
- Add validations to Flat
- Add validations to Reservation
- Construct routes

### CODE REVIEW

File: app/models/flat.rb [hunk 1]
```diff
@@ -1,3 +1,9 @@
 class Flat < ApplicationRecord
   belongs_to :user
+  has_many :reservations
+  has_many :users, through: :reservations
+  validates :name, presence: true
+  validates :address, presence: true
+  validates :description, presence: true
+  validates :price, presence: true
```
Comment (pontusnordbrandt) [inline app/models/flat.rb:8]: Does this need: , numericality: { only_integer: true }

Comment (pontusnordbrandt) [reply app/models/flat.rb:8]: ?

Comment (reba-g) [reply app/models/flat.rb:8]: ok cool!

File: app/models/user.rb [hunk 2]
```diff
@@ -1,2 +1,6 @@
 class User < ApplicationRecord
+  has_many :flats, dependent: :destroy
+  has_many :reservations
+  validates :name, presence: true
+  validates :email, presence: true
```
Comment (pontusnordbrandt) [inline app/models/user.rb:5]: uniqueness: true

File: app/models/flat.rb (hunk 1 above)
Comment (reba-g) [reply app/models/flat.rb:8]: i think it should actually be numericality: true so it can be a float

**Pull request closed without merging**